### DISCUSSION TITLE
sudo so-rule modify is not working

### DISCUSSION THREAD

**Discussion by maiwandh** [Security-Onion-Solutions/securityonion#6137] [Unsupported Versions] [answered]
I´m new to security onion 2.3 and I´m trying to modify the existing rulesets to play around a little bit.

I tried to change the Variable **"$EXTERNAL_NET"** to **"$HOME_NET"** with the following command.
`sudo so-rule modify add 2009582 "\$EXTERNAL_NET" "\$HOME_NET"` (also tried with only "$HOME_NET"

I just wan´t to catch with the following nmap scan when it is performed from a machine inside my home network: `nmap -sS`

I can see the change with sudo so-rule modify list as follows:
![image](https://user-images.githubusercontent.com/43877044/140747555-a90a1e4c-f3f2-4f9d-aa70-d9ad74464fe6.png)

But when I check the rull in the all.rules file I see that nothing have changed. I´m not sure what I´m doing wrong :S
![image](https://user-images.githubusercontent.com/43877044/140747577-77e3f2b4-4bc1-4e8b-87a1-917193d76950.png)

**Comment by dougburks** [comment, marked as answer]
Looks like the rule is commented out (disabled). Try enabling it to see if your modification takes effect.

**Reply by maiwandh** [reply]
Thanks for your response @dougburks 
Unfortunately this doesn´t work, too.

I enabled the rule first, then removed the existing modifications on the rule and after that I added the modification again.

Now with the enabled rule I got still the following output:
![image](https://user-images.githubusercontent.com/43877044/141117586-5d0da39c-1965-4e72-8dd9-381827c8c975.png)

**Reply by maiwandh** [reply]
I also noticed that it is e.g. possible to modify the disabled or enabled rule in the msg option. It seems for me that it´s maybe not possible to change the ruleheader? Or should this be possible?

**Reply by rwwiv** [reply]
@maiwandh the example given by `so-rule` is actually incorrect, to include an escaped `$` character in the regex pattern you'll actually want to use the following command:
```sh
sudo so-rule modify add 2009582 "\\\$EXTERNAL_NET" "\$HOME_NET"
# results in
# 2009582 "\$EXTERNAL_NET" "$HOME_NET"
```

N.B. The first string is actually a regex pattern. the second is just a raw value. You'll need to ensure the first of the two properly escapes any characters that would be interpreted by regex, the second only needs the `$` character escaped to prevent bash from treating that as a variable.

**Reply by maiwandh** [reply]
@rwwiv Thank you very much sir. That solved my problem.

**Reply by dougburks** [reply]
Added to documentation:
https://docs.securityonion.net/en/2.3/managing-alerts.html#modify-the-sid
